src/misc/programs.cpp

```diff
@@ -133,8 +133,30 @@ void Program::ChangeToLongCmd() {
 	full_arguments.assign(""); //Clear so it gets even more save
 }
 
-static char last_written_character = 0;//For 0xA to OxD 0xA expansion
-void Program::WriteOut(const char * format,...) {
+bool Program::SuppressWriteOut(const char *format)
+{
+	// Have we encountered an executable thus far?
+	static bool encountered_executable = false;
+
+	// Should we consider suppressing?
+	const bool consider_suppressing = control->GetStartupVerbosity() ==
+	                                          Verbosity::Quiet &&
+	                                  control->cmdline->HasExecutable();
+
+	if (encountered_executable || !consider_suppressing)
+		return false;
```

Comment:
Code is ok, but we could avoid lookup inside `control` completely, and perhaps we should prepare for scenario, that in the future there might be verbosity lower than Quiet?

How about:
```suggestion
	if (encountered_executable)
		return false;
	if (control->GetStartupVerbosity() > Verbosity::Quiet)
		return false;
	if (!control->cmdline->HasExecutable())
		return false;
```

Reply:
Very nice! Added.